appium/webdriver/extensions/android/activities.py

```diff
@@ -79,9 +79,11 @@ def current_activity(self) -> str:
         ext_name = 'mobile: getCurrentActivity'
         try:
             return self.assert_extension_exists(ext_name).execute_script(ext_name)
-        except UnknownMethodException:
-            # TODO: Remove the fallback
-            return self.mark_extension_absence(ext_name).execute(Command.GET_CURRENT_ACTIVITY)['value']
+        except WebDriverException as e:
+            if isinstance(e, UnknownMethodException) or self.is_missing_command(e.msg):
```

Comment:
could it be the server throws anything else other than UnknownMethodException in case of missing method?

Reply:
This must be a bug in the client itself. I believe it does not properly map the exception raised by the server because this works as expected with Java

Reply:
No worries, I have already prepared a fix: https://github.com/appium/python-client/pull/869

Thanks for pointing this out, I did not know it's so broken. Seems like the actual error handling code in selenium client has not been updated since ages and only supports responses in JWP format

Reply:
thanks, gotcha. Then this pr is no longer needed